src/cmd/init.rs

```diff
@@ -199,21 +189,96 @@ pub fn run_init(wallet_dir: &std::path::Path, cmd: InitOpt) -> anyhow::Result<Cm
                     let sw_file = cmd::get_seed_words_file(wallet_dir);
                     fs::write(sw_file.clone(), seed_words.clone())?;
                     eprintln!("Wrote seeds words to {}", sw_file.display());
-                    WalletKey::SeedWordsFile {}
+                    (sw_file, seed_words)
+                }
+            };
+
+            let mnemonic = Mnemonic::parse(&seed_words).map_err(|e| {
+                anyhow!(
+                    "parsing seed phrase in '{}' failed: {}",
+                    sw_file.as_path().display(),
+                    e
+                )
+            })?;
+
+            let passphrase = if has_passphrase {
+                eprintln!("Warning: by using a passphrase you are mutating the secret derived from your seed words. \
+                If you lose or forget your seedphrase, you will lose access to your funds.");
+                loop {
+                    let passphrase =
+                        rpassword::prompt_password_stderr("Please enter your wallet passphrase: ")?;
+                    let passphrase_confirmation = rpassword::prompt_password_stderr(
+                        "Please confirm your wallet passphrase: ",
+                    )?;
+                    if !passphrase.eq(&passphrase_confirmation) {
+                        eprintln!("Mismatching passphrases. Try again.\n")
+                    } else {
+                        break passphrase;
+                    }
                 }
+            } else {
+                "".to_string()
             };
+
+            let seed_bytes = mnemonic.to_seed(passphrase);
+            // Create secret randomness from seed.
+            let xpriv = ExtendedPrivKey::new_master(common_args.network, &seed_bytes).unwrap();
+            let secp = Secp256k1::signing_only();
+            let bip85_key = xpriv
+                .derive_priv(
+                    &secp,
+                    &DerivationPath::from_str("m/83696968'/128169'/64'/330'").unwrap(),
+                )
+                .unwrap();
+
+            let message = hex::decode(&format!("{}", bip85_key.private_key.display_secret()))?;
+            let mut engine = HmacEngine::<sha512::Hash>::new("bip-entropy-from-k".as_bytes());
+            engine.input(&message);
+            let hash = Hmac::<sha512::Hash>::from_engine(engine);
```

Comment:
:wrench:  Put this derivation into its own function and write a test against the BIP please.